Error building master on Ubuntu (file cannot create directory: /usr/local/share/man/man1)

When invoking ``make install``, I'm getting the error ``file cannot create directory: /usr/local/share/man/man1``, but that's outside the specified install prefix, so the build script should not try to write to it.

(This used to work, I don't know exactly at which point it stopped working)

- Operating system/version: 

```
％ uname -a
Linux xkuntu 5.0.0-20-generic #21-Ubuntu SMP Mon Jun 24 09:32:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
％ cat /etc/*-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=19.04
DISTRIB_CODENAME=disco
DISTRIB_DESCRIPTION="Ubuntu 19.04"
NAME="Ubuntu"
VERSION="19.04 (Disco Dingo)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 19.04"
VERSION_ID="19.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=disco
UBUNTU_CODENAME=disco
```


### Steps to reproduce

```
% git clone https://github.com/neovim/neovim.git /tmp/neovim
% pushd /tmp/neovim
% make CMAKE_BUILD_TYPE=RelWithDebInfo CMAKE_INSTALL_PREFIX=/tmp/neovim-prefix CMAKE_EXTRA_FLAGS="-DCMAKE_FIND_FRAMEWORK=LAST" BUNDLED_CMAKE_FLAG="-DUSE_BUNDLED_GPERF=OFF"
[ succeeds ]
％ make install

```

### Actual behavior

```
mkdir -p build
touch build/.ran-third-party-cmake
ninja -C .deps
ninja: Entering directory `.deps'
[2/2] cd /tmp/neovim/.deps && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.14.5/bin/cmake -E touch .third-party
ninja -C build
ninja: Entering directory `build'
ninja: no work to do.
ninja -C build install
ninja: Entering directory `build'
[0/1] Install the project...
-- Install configuration: "RelWithDebInfo"
CMake Error at cmake_install.cmake:77 (file):
  file cannot create directory: /usr/local/share/man/man1.  Maybe need
  administrative privileges.


FAILED: CMakeFiles/install.util
cd /tmp/neovim/build && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.14.5/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
make: *** [Makefile:159: install] Error 1
```

### Expected behavior

neovim compiles and installs in the provided prefix :D

--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ CMAKE_INSTALL_PREFIX := $(shell echo $(CMAKE_EXTRA_FLAGS) | 2>/dev/null \
     grep -o 'CMAKE_INSTALL_PREFIX=[^ ]\+' | cut -d '=' -f2)
 endif
 ifneq (,$(CMAKE_INSTALL_PREFIX))
-CMAKE_EXTRA_FLAGS += -DCMAKE_INSTALL_PREFIX=$(CMAKE_INSTALL_PREFIX)
+override CMAKE_EXTRA_FLAGS += -DCMAKE_INSTALL_PREFIX=$(CMAKE_INSTALL_PREFIX)
 
 checkprefix:
 	@if [ -f build/.ran-cmake ]; then \

